libraries/AP_Baro/AP_Baro_SPL06.cpp

```diff
@@ -108,21 +115,44 @@ bool AP_Baro_SPL06::_init()
 
 // Sometimes SPL06 has init problems, that's due to failure of reading using SPI for the first time. The SPL06 is a dual
 // protocol sensor(I2C and SPI), sometimes it takes one SPI operation to convert it to SPI mode after it starts up.
-    bool is_SPL06 = false;
 
     for (uint8_t i=0; i<5; i++) {
-        if (_dev->read_registers(SPL06_REG_CHIP_ID, &whoami, 1)  &&
-            whoami == SPL06_CHIP_ID) {
-            is_SPL06=true;
-            break;
+        if (_dev->read_registers(SPL06_REG_CHIP_ID, &whoami, 1)) {
+        	switch(whoami) {
+			case SPL06_CHIP_ID:
+				type = Type::SPL06;
+				break;
+			case SPA06_CHIP_ID:
+				type = Type::SPA06;
+				break;
+			default:
+				type = Type::UNKNOWN;
+				break;
+			}
         }
+
+        if (type != Type::UNKNOWN)
+			break;
     }
     
-    if(!is_SPL06) {
+    if (type == Type::UNKNOWN) {
         return false;
     }
 
     // read the calibration data
+    uint8_t SPL06_CALIB_COEFFS_LEN;
+	switch(type) {
+	case Type::SPL06:
+		SPL06_CALIB_COEFFS_LEN = SPL06_REG_CALIB_COEFFS_END - SPL06_REG_CALIB_COEFFS_START + 1;
+		break;
+	case Type::SPA06:
+		SPL06_CALIB_COEFFS_LEN = SPA06_REG_CALIB_COEFFS_END - SPL06_REG_CALIB_COEFFS_START + 1;
+		break;
+	default:
+		SPL06_CALIB_COEFFS_LEN = 18;
+		break;
```

Comment:
```suggestion
```

by omitting the default case we help someone adding the next `Type::SPx06` driver as they're forced to fill in each of the switch statements.